From c7c6bcbcb0c51b4b290ef127bb51792237e5e7d0 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:00 -0400
Subject: [tools] perf/machine: Fix id_hdr_size initialization

Message-id: <1408953210-26343-233-git-send-email-jolsa@redhat.com>
Patchwork-id: 88965
O-Subject: [PATCH RHEL7.1 BZ1133083 232/322] perf machine: Fix id_hdr_size initialization
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 14bd6d20fef603060474701967085442b716b6a9
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Jan 7 13:47:19 2014 +0100

description
===========
The id_hdr_size field was not properly initialized, set it to zero, as
the machine struct may have come from some non zeroing allocation
routine or from the stack without any field being initialized.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/machine.c b/tools/perf/util/machine.c
index a98538d..0130279 100644
--- a/tools/perf/util/machine.c
+++ b/tools/perf/util/machine.c
@@ -27,6 +27,7 @@ int machine__init(struct machine *machine, const char *root_dir, pid_t pid)
  machine->pid = pid;
 
  machine->symbol_filter = NULL;
+ machine->id_hdr_size = 0;
 
  machine->root_dir = strdup(root_dir);
  if (machine->root_dir == NULL)
-- 
1.7.1